src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/CpxVariantCanonicalRepresentation.java

```diff
@@ -447,8 +467,12 @@ private CpxVariantCanonicalRepresentation(final Kryo kryo, final Input input) {
 
         final int numSegments = input.readInt();
         referenceSegments = new ArrayList<>(numSegments);
-        for (int i = 0; i < numSegments; ++i)
-            referenceSegments.add(kryo.readObject(input, SimpleInterval.class));
+        for (int i = 0; i < numSegments; ++i) {
```

Comment:
Why did you have to re-do this serialization instead of delegating to `SimpleInterval.Serializer`?

Reply:
There was no Kryo serializer for `SimpleInterval` since it is Java serializable, and the class is not registered either.

The serialization tests (which are new in this PR) fail if I did the `kryo.readObject(input, SimpleInterval.class)`